delta/kernel/src/main/scala/ch/epfl/bluebrain/nexus/delta/kernel/utils/IOUtils.scala

```diff
@@ -22,6 +23,13 @@ object IOUtils extends IOUtils
 trait IOInstant {
   def now(implicit clock: Clock[IO]): IO[Instant] =
     clock.realTime(TimeUnit.MILLISECONDS).map(Instant.ofEpochMilli)
+
+  def timed[A](io: IO[A])(implicit c: Clock[IO]): IO[(A, FiniteDuration)] =
+    for {
+      start  <- c.monotonic(MILLISECONDS)
+      result <- io
+      finish <- c.monotonic(MILLISECONDS)
+    } yield (result, (finish - start).millis)
```

Comment:
Probably worth a test

Reply:
I'm reluctant since it's essentially copied stdlib code which you wouldn't normally test... Kinda covered in those slow query tests as well

Reply:
I think copied code is still worth testing if we rely on it. Tests are there to make sure it works going forward as well as now. It should be super easy to write